sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/models/AmqpAnnotatedMessageTest.java

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Test class for  {@link AmqpAnnotatedMessage}
+ */
+public class AmqpAnnotatedMessageTest {
+
+    private static final byte[] CONTENTS_BYTES = "Some-contents".getBytes(StandardCharsets.UTF_8);
+    private static final BinaryData DATA_BYTES = new BinaryData(CONTENTS_BYTES);
+
+    /**
+     * Verifies we correctly set values via constructor for {@link AmqpAnnotatedMessage}.
+     */
+    @Test
+    public void constructorValidValues() {
+        // Arrange
+        final List<BinaryData> binaryDataList = Collections.singletonList(DATA_BYTES);
+        final AmqpDataBody amqpDataBody = new AmqpDataBody(binaryDataList);
+
+        // Act
+        AmqpAnnotatedMessage actual = new AmqpAnnotatedMessage(amqpDataBody);
+
+        // Assert
+        assertMessageCreation(actual, AmqpBodyType.DATA, binaryDataList.size());
+    }
+
+    /**
+     * Verifies we correctly set values via constructor for {@link AmqpAnnotatedMessage}.
+     */
+    @Test
+    public void constructorAmqpValidValues() {
+        // Arrange
+        final List<BinaryData> listBinaryData = Collections.singletonList(DATA_BYTES);
+        final AmqpDataBody amqpDataBody = new AmqpDataBody(listBinaryData);
+        AmqpAnnotatedMessage expected = new AmqpAnnotatedMessage(amqpDataBody);
```

Comment:
in general, I'd be consistent about when you're using "final" or not. This applies to all other areas of your code.